### PR TITLE
test: verify reviewbot unified review (throwaway)

### PR DESCRIPTION
**Pull request by JSONbored**
Throwaway PR to verify reviewbot's **unified CodeRabbit-style review** in production: one review = summary body + Changes table + consolidated `Prompt for AI agents` + inline committable suggestions under it. The `scratch/` fixture is outside CI scope and has two latent issues for the reviewer to find. Will be closed — **do not merge**.

### CODE REVIEW

**Comment by ghost** [comment]
> [!NOTE]
> ## Gittensory Gate skipped
> PR closed before full evaluation. No late first comment was created.
>
> | Signal | Result | Evidence | Action |
> | --- | --- | --- | --- |
> | Gate result | ⚠️ Skipped | JSONbored/gittensory#630 is no longer open. | No action. |

---
💰 **Earn for open-source contributions like this.** [Gittensor](https://gittensor.io) lets GitHub contributors earn for the work they already do — [register to start earning →](https://gittensor.io/miners/repository?name=JSONbored%2Fgittensory&tab=miners).

Checked by [Gittensory](https://gittensory.aethereal.dev), a quiet PR intelligence layer for OSS maintainers.

**Review by ghost — COMMENTED** [review]
**Gittensory review** · _advisory — a maintainer merges_

**0** actionable · **0** nitpick(s) · 1 file(s) · two independent AI reviewers

> **Suggested action:** ✅ **Safe to merge** — both reviewers found no blocking issues.

<details>
<summary>📋 Walkthrough</summary>

Adds `scratch/reviewbot-unified-demo.ts` with a `mean` function to compute the average of an array of numbers and a `parsePort` function to parse a port string with a default of 8080. The file is explicitly marked as outside the production code path.

**Changes**

| File | Summary |
| --- | --- |
| `scratch/reviewbot-unified-demo.ts` | Added a new scratch file with two simple utility functions. |

</details>

<details>
<summary>🔍 Reviewer notes</summary>

**Reviewer A · `gpt-oss-120b`** — recommends ✅ `merge`
The PR adds a new throwaway TypeScript file in the scratch directory containing two simple utility functions. The change is isolated, does not affect the build or tests, and introduces no functional impact.

_No blocking issues spotted._

**Reviewer B · `nemotron-3-120b-a12b`** — recommends ✅ `merge`
The PR adds a throwaway test file in the scratch directory to verify the reviewbot's unified review functionality. The file contains two simple utility functions (mean and parsePort) and is explicitly outside the build, typecheck, test, and coverage scope, making it safe and without impact on the main codebase.

_No blocking issues spotted._

</details>

---
💰 Earn for open-source contributions like this — [Gittensor](https://gittensor.io) lets contributors earn for the work they already do. · _Advisory review by Gittensory._

File: scratch/reviewbot-unified-demo.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Small numeric helpers.
+
+/** Clamp a percentage into the 0–100 range. */
+export function clampPercent(value: number): number {
+  if (value > 100) {
```
Comment (ghost) [inline scratch/reviewbot-unified-demo.ts:5]: **⚠️ Potential issue** · Missing lower bound clamp

`clampPercent` only caps values above 100 but does not prevent negative percentages from being returned.

```suggestion
if (value < 0) { return 0; }
```

<details>
<summary>🤖 Prompt for AI agents</summary>

```
In scratch/reviewbot-unified-demo.ts around line 5, `clampPercent` only caps values above 100 but does not prevent negative percentages from being returned. Apply: if (value < 0) { return 0; }
```

</details>

**Review by ghost — COMMENTED** [review]
**Gittensory review** · _advisory — a maintainer merges_

**2** actionable · **5** nitpick(s) · 1 file(s) · two independent AI reviewers

> **Suggested action:** 🛠️ **Request changes.** _(reviewers split: request changes / merge)_ Address the suggestions below before merging.

<details>
<summary>📋 Walkthrough</summary>

Adds `scratch/reviewbot-unified-demo.ts` containing three numeric helpers: clampPercent, average, and parsePort. The functions lack full validation for edge cases.

**Changes**

| File | Summary |
| --- | --- |
| `scratch/reviewbot-unified-demo.ts` | Introduces three numeric helper functions (clampPercent, average, parsePort). |

</details>

<details>
<summary>🔍 Reviewer notes</summary>

**Reviewer A · `gpt-oss-120b`** — recommends 🛠️ `request changes`
The PR adds a new utility file with three simple numeric helper functions. The code compiles but has edge‑case bugs (no lower‑bound clamp, average on empty arrays, and no validation of parsed ports).

_Suggestions_
- Clamp negative values to 0 in `clampPercent`.
- Guard against empty input in `average` to avoid division by zero.
- Validate that `parsePort` returns a positive integer within the valid port range (1‑65535).
- Add unit tests for these helpers covering edge cases.

_Worth double-checking_
- Runtime NaN from `average([])` could propagate errors.
- `clampPercent` returning negative numbers may break UI percent displays.
- Invalid port numbers could cause server startup failures.

**Reviewer B · `nemotron-3-120b-a12b`** — recommends ✅ `merge`
The PR adds a small utility file with three pure TypeScript helper functions (clampPercent, average, parsePort) in a scratch directory. The code is straightforward, type‑safe, and introduces no side effects or public‑facing changes.

_No blocking issues spotted._

</details>

<details>
<summary>🤖 Prompt for AI agents</summary>

Copy this to your coding agent to fix everything flagged:

```
Address the following in this PR:

1. scratch/reviewbot-unified-demo.ts (line 5): `clampPercent` only caps values above 100 but does not prevent negative percentages from being returned. Suggested fix: if (value < 0) { return 0; }
2. scratch/reviewbot-unified-demo.ts (line 13): `average` divides by `values.length` without checking for an empty array, yielding NaN. Suggested fix: if (values.length === 0) { return 0; }
3. scratch/reviewbot-unified-demo.ts (line 18): `parsePort` returns any numeric value, including negatives or >65535, which are invalid ports. Suggested fix: const port = parseInt(raw); return (port > 0 && port <= 65535) ? port : 8080;

Nitpicks:
- Clamp negative values to 0 in `clampPercent`.
- Guard against empty input in `average` to avoid division by zero.
- Validate that `parsePort` returns a positive integer within the valid port range (1‑65535).
- Add unit tests for these helpers covering edge cases.
```

</details>

---
_🤖 Automated advisory review · a maintainer makes the final call._

File: scratch/reviewbot-unified-demo.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    return 100;
+  }
+  return value;
+}
+
+/** Return the average of the numbers. */
+export function average(values: number[]): number {
+  return values.reduce((a, b) => a + b, 0) / values.length;
```
Comment (ghost) [inline scratch/reviewbot-unified-demo.ts:13]: **⚠️ Potential issue** · Division by zero on empty array

`average` divides by `values.length` without checking for an empty array, yielding NaN.

```suggestion
if (values.length === 0) { return 0; }
```

<details>
<summary>🤖 Prompt for AI agents</summary>

```
In scratch/reviewbot-unified-demo.ts around line 13, `average` divides by `values.length` without checking for an empty array, yielding NaN. Apply: if (values.length === 0) { return 0; }
```

</details>

File: scratch/reviewbot-unified-demo.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+/** Parse a port number from a string, defaulting to 8080. */
+export function parsePort(raw: string): number {
+  return parseInt(raw) || 8080;
```
Comment (ghost) [inline scratch/reviewbot-unified-demo.ts:18]: **🧹 Nitpick** · Port validation

`parsePort` returns any numeric value, including negatives or >65535, which are invalid ports.

```suggestion
const port = parseInt(raw); return (port > 0 && port <= 65535) ? port : 8080;
```

<details>
<summary>🤖 Prompt for AI agents</summary>

```
In scratch/reviewbot-unified-demo.ts around line 18, `parsePort` returns any numeric value, including negatives or >65535, which are invalid ports. Apply: const port = parseInt(raw); return (port > 0 && port <= 65535) ? port : 8080;
```

</details>

**Comment by JSONbored** [comment]
Closing — throwaway test PR for verifying reviewbot's review UX. Thanks, done.
